### PR TITLE
Add a roundrobin connector, that can help single thread components to scale

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

**Review by dmitryax — APPROVED** [review]
Nice! Pretty straight-forward implementation

File: connector/roundrobinconnector/connector.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		}
+		consumers[i] = cons
+	}
+	return consumers, nil
+}
+
+type router[T any] interface {
+	PipelineIDs() []component.ID
+	Consumer(pipelineIDs ...component.ID) (T, error)
+}
```
Comment (djaglowski) [inline connector/roundrobinconnector/connector.go:34]: Should this be how we implement it in core?

**Review by djaglowski — APPROVED** [review]
LGTM

File: connector/roundrobinconnector/connector.go (hunk 1 above)
Comment (bogdandrutu) [reply connector/roundrobinconnector/connector.go:34]: You mean to help remove duplicate code? I can craft a quick PR.

**Pull request merged**